lib/ibm_cloud_sdk_core/jwt_token_manager.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require("http")
+require("json")
+require("jwt")
+require("rbconfig")
+require_relative("./version.rb")
+require_relative("./jwt_token_manager")
+
+TOKEN_NAME = "access_token"
+
+module IBMCloudSdkCore
+  # Class to manage JWT Token Authentication
+  class JWTTokenManager
+    def initialize(vars)
+      defaults = {
+        token_info: nil,
+        url: nil,
+        access_token: nil
+      }
+      vars = defaults.merge(vars)
+
+      @url = vars[:url]
+      @token_info = vars[:token_info]
+      @user_access_token = vars[:access_token]
+      @time_to_live = nil
+      @expire_time = nil
+      @disable_ssl_verification = false
+    end
+
+    def token
+      if !@user_access_token.nil?
+        @user_access_token
+      elsif @token_info.nil? || token_expired?
+        token_info = request_token
+        save_token_info(token_info: token_info)
+        @token_info[TOKEN_NAME]
+      end
+    end
```

Comment:
I still feel like this might be missing something. Are you using the same variable for both user-managed tokens and SDK-managed tokens? If so, this is correct and good to go 👍 . If not, there needs to be an option to return the saved, SDK-managed token from token info. I think the other SDKs use an `else` statement at the end of this check.

Reply:
Good catch, it's missing the option when `@token_info` is not nil. we should be returning the token in that case. I will update the logic.

Reply:
Updated the logic!